packages/Input/src/Input.styles.js

```diff
@@ -19,7 +19,7 @@ const iconStyles = {
   `,
 };
 
-const inputStyles = props => `
+const inputStyles = ({ hasClearButton, size }) => `
```

Comment:
this seems a little off, wouldn't make more sense to use 
```js
const inputStyles = css`
   ${props => {
        props.hasClearButon; props.size;
     }}
`
```

Reply:
I don't quite follow your example, but find the way I have it keeps the subsequent conditions cleaner (no need for the `props =>` or `props.` anywhere).  Also, kinda nice to see what props these styles depend on right at the top in a convenient location.

Reply:
One possible downside of using as a function that creates css is that will block any static analyzers to detect css code, like this [plugin](https://github.com/siffogh/eslint-plugin-better-styled-components).

See example: 

![2019-06-21 11 54 11](https://user-images.githubusercontent.com/19418590/59945225-d039fd80-941b-11e9-9367-bb6399fdafca.gif)

While I like and preferer your approach or using a function instead of **css``**, for the benefit of the entire repo is better to use **css``** IMHO.

What do you think?

Reply:
While I haven't been able to get syntax highlighting to work properly with the `css**` notation either, if that's something I _could_ and this is the blocker, I don't mind compromising.  I'll probably keep destructuring `props` though, I think it's still a little better:
```js
${({ isDragging }) => (isDragging ? draggingStyles : null)}
```

Reply:
destructuring is better agree, and at least in atom unless I follow the pattern of `styled.div''` or `css''` highlighting css will not work.

Reply:
Yep. We should probably create an issue to refactor components to use this pattern consistently.

Reply:
FYI https://github.com/acl-services/paprika/issues/98 @mikrotron